### ISSUE TITLE
Not working for checking non-finite loss files ? 

### ISSUE THREAD

**Issue by MuruganR96** [mozilla/DeepSpeech#2474]
### checking non-finite loss files

Thanks for added this feature in DeepSpeech latest branch.

I tried this concept for v0.5.1 master and i modified lightly for feeding.py, DeepSpeech.py and evaluate.py.

**feeding.py**
```
def batch_fn(wav_filenames, features, features_len, transcripts):
        wav_filenames = wav_filenames.batch(batch_size)
        return tf.data.Dataset.zip((wav_filenames, features, transcripts))
```
**DeepSpeech.py**
```
non_finite_files = tf.gather(batch_filenames, fv1.where(~tf.math.is_finite(total_loss)))
```

- **Have I written custom code (as opposed to running examples on an unmodified clone of the repository)**: Yes
- **OS Platform and Distribution:** Linux Ubuntu 16.04
- **TensorFlow version (use command below)**: 1.13.1
- **Python version**: 3.6.5
- **CUDA/cuDNN version**:Cuda 10.0

- **GPU model and memory**: 4  x 24 GB TITAN RTX

You can obtain the TensorFlow version with

```bash
python -c "import tensorflow as tf; print(tf.GIT_VERSION, tf.VERSION)"

b'v1.13.1-0-g6612da8951' 1.13.1

```

### training loss and validations giving high but why it is not identify non-finite-loss files (wrong transcripts files)? 

if i did any mistakes, please correct me. can someone explain me how it is working actually?

Thank you for all.

**Comment by lissyx** [comment]
@MuruganR96 It's a bit unclear in the current form. This change makes it possible to know which files has infinite loss ? I'm sure we already have something for that on master.

**Comment by kdavis-mozilla** [comment]
@lissyx I think he's trying to backport that to 0.5.1?

**Comment by MuruganR96** [comment]
@lissyx  @kdavis-mozilla Thank you sir.

yes, i am trying to backport 0.5.1. but i can't able to get the list of non finite loss files. 
am i miss something,  could you explain me?

**Comment by lissyx** [comment]
> but i can't able to get the list of non finite loss files.

Please be more specific.

**Comment by kdavis-mozilla** [comment]
@MuruganR96 As this is not a bug with master, 0.6.X, or any older release version, I suggest this should be moved to [Discourse](https://discourse.mozilla.org/c/deep-speech).

**Comment by MuruganR96** [comment]
@lissyx  sir.  
>  I'm sure we already have something for that on master.
> > but i can't able to get the list of non finite loss files.
> 
> Please be more specific.

in DeepSpeech v0.5.1 not implemented for **checking non-finite loss files**.

so what i did here is, just i applied same concepts in 0.5.1. but i am not getting problem files(non finite files).

in my 250 hrs of datasets i trained in 0.5.1. after few epochs after validation went **inf loss**.
so if i will implement checking non-finite loss files mean, i can resolve this inf loss issue via removing the problem files.

thank you sir.

**Comment by MuruganR96** [comment]
> @MuruganR96 As this is not a bug with master, 0.6.X, or any older release version, I suggest this should be moved to [Discourse](https://discourse.mozilla.org/c/deep-speech).

Thank you sir. Please discuss here sir [My issue](https://discourse.mozilla.org/t/not-working-for-checking-non-finite-loss-files/47627?u=muruganrajenthirean)

**Comment by lissyx** [comment]
> so what i did here is, just i applied same concepts in 0.5.1. but i am not getting problem files(non finite files).

If you continue not to share your patch, it's going to be hard to try and help you.

Why don't you just use master ?

**Comment by MuruganR96** [comment]
> If you continue not to share your patch, it's going to be hard to try and help you.
> 
> Why don't you just use master ?

Master v0.5.1 is the latest one upto my knowledge ?  here did they already implemented **checking non-finite loss files** ?

**Comment by lissyx** [comment]
> Master v0.5.1 is the latest one upto my knowledge

Please be clear, you are mixing everything here.

**Comment by MuruganR96** [comment]
@lissyx sir. 

in my datasets 250 hrs duration.

like youtube datasets segmented with corresponding subtitle transcript.
here some spelling mistakes and mispronunciations files is there.

datasets i was trained in 0.5.1. few epochs after validation went inf loss.

so what i thought is, worst case simply removing the file from your dataset.

so i need to find out problem files.  that's why i am focusing on checking non-finite-loss files to remove sir. 

@lissyx Thank you sir.
